src/edit_view.rs

```diff
@@ -59,16 +60,20 @@ pub enum EditViewCommands {
     SelectAll,
 }
 
+type Method = String;
+type Params = Value;
+
 /// State and behavior for one editor view.
 pub struct EditView {
-    view_id: String,
+    view_id: Option<String>,
     line_cache: LineCache,
     dwrite_factory: directwrite::Factory,
     resources: Option<Resources>,
     scroll_offset: f32,
     size: (f32, f32),  // in px units
     viewport: Range<usize>,
     core: Weak<Mutex<Core>>,
+    pending: RefCell<Vec<(Method, Params)>>,
```

Comment:
This can be done without RefCell. Here's your game plan:

* Make `char`, `send_edit_cmd`, and `send_action` take `&mut self`. The fact they were `&self` is an artifact of the way wndproc needs to be reentrant in general; in druid widgets get mutable access to their state and the ui toolkit takes care of the refcell stuff.

* Drain pending with `mem::replace(&mut self.pending,
Vec::new())`, that way you don't need to hold a mutable reference while applying the pending commands.